common/TextDifferencer.java

```diff
@@ -51,7 +52,34 @@
     allChanges.addAll(
         getMatchingTextChanges(
             first, first.length - footerLength, second.length - footerLength, footerLength));
-    return allChanges;
+    return unifyTextChanges(ImmutableList.copyOf(allChanges));
+  }
+
+  /** Unifies TextChanges from chars into strings  */
+  private static ImmutableList<TextChange> unifyTextChanges(ImmutableList<TextChange> changes) {
+    ImmutableList.Builder<TextChange> result = new ImmutableList.Builder<>();
+    TextChange unified = null;
+    TextChange previous = null;
+    for (TextChange current : changes) {
+      System.out.println(current);
+      if (previous == null) {
+        previous = current;
+        unified = current;
+        continue;
+      }
+      if (previous.getType() == current.getType()) {
+        unified = unified.toBuilder().setDifference(unified.getDifference() + current.getDifference()).build();
```

Comment:
If you will work with builders directly, you can avoid changing back and forth from builder to protobuf.

Reply:
Done